src/ortho.rs

```diff
@@ -589,4 +659,469 @@ mod tests {
         norms2.sort();
         assert_eq!(norms1, norms2, "Canonicalization mismatch between axis insertion orders. norms1={:?} norms2={:?}", norms1, norms2);
     }
+
+    #[test]
+    fn test_rebuild_places_impacted_key_furthest_right() {
+        // Prove that when rebuilding an ortho via subtract, the last remaining key
+        // ends up at the "most advanced position" (furthest right, ready for next add)
+        
+        // Build an ortho with values [10, 20, 30, 40]
+        let ortho = Ortho::new(1);
+        let ortho = ortho.add(10, 1)[0].clone();
+        let ortho = ortho.add(20, 1)[0].clone();
+        let ortho = ortho.add(30, 1)[0].clone();
+        let ortho = ortho.add(40, 1)[0].clone();
+        
+        // Current position should be 4 (all 4 slots filled in base [2,2])
+        assert_eq!(ortho.get_current_position(), 4);
+        
+        // Subtract once - removes 40, leaving [10, 20, 30]
+        let sub1 = ortho.subtract().expect("Should subtract");
```

Comment:
For rebuild scenarios do not frame it as subtract. It is simply a rebuild up to a certain change point, and having the right geometry. Prove this through an example going down a search tree and proving you can rebuild to any point in it